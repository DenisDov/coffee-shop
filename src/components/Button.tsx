import * as Haptics from "expo-haptics";
import React from "react";
import { Pressable, StyleSheet } from "react-native";

import { Text, theme } from "@/theme";

type Props = {
  onPress: () => void;
  title?: string;
};

const RIPPLE_CONFIG = {
  color: theme.colors.primary,
  borderless: true,
};

export const Button = ({ onPress, title }: Props) => {
  const handlePress = () => {
    Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
    onPress();
  };
  return (
    <Pressable
      onPress={handlePress}
      hitSlop={16}
      android_ripple={RIPPLE_CONFIG}
      style={({ pressed }) => [
        styles.button,
        {
          opacity: pressed ? 0.7 : 1,
        },
      ]}
    >
      <Text variant="button">{title}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: theme.colors.primary,
    paddingHorizontal: theme.spacing.s,
    height: theme.spacing.xxl,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: theme.borderRadii.m,
  },
});
