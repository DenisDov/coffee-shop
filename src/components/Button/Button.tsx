import React from 'react';
import { StyleSheet } from 'react-native';
import { RectButton } from 'react-native-gesture-handler';

import { Box, Text, theme } from '@/theme';

export type ButtonProps = {
  onPress: () => void;
  text?: string;
};

export const Button = ({ onPress, text }: ButtonProps) => {
  return (
    <RectButton onPress={onPress} hitSlop={16} style={styles.button}>
      <Box accessible accessibilityRole="button">
        <Text variant="button">{text}</Text>
      </Box>
    </RectButton>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: theme.colors.primary,
    paddingHorizontal: theme.spacing.s,
    height: theme.spacing.xxl,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: theme.borderRadii.m,
  },
});
