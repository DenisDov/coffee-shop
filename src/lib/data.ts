const categories: string[] = [
  'Black',
  'Espresso',
  'Cappuccino',
  'Macchiato',
  'Latte',
  'Americano',
  'Ristretto',
  'Flat white',
  'Doppio',
];

const coffees = [
  {
    id: '1',
    title: 'Cappuccino',
    extras: 'Chocolate',
    description:
      'The cappuccino, a classic Italian coffee, embodies a harmonious blend of espresso, steamed milk, and a luscious layer of milk foam. Originating in Italy, this beloved beverage has become a global favorite. With its precise ratio of one-third espresso, one-third steamed milk, and one-third foam, the cappuccino delivers a balanced, rich flavor profile. The robust espresso forms the foundation, offering a bold and intense kick, harmonized by the velvety smoothness of the steamed milk. Topped with a frothy crown of foam, each sip of cappuccino presents a delightful contrast of flavors and textures, inviting coffee enthusiasts into a realm of pure indulgence and comfort.',
    price: 4.49,
    image: {
      source:
        'https://images.unsplash.com/photo-1472973681244-f5bcc808ad47?q=80&w=2974&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: '1zgKDYJ2d5aviHdmerhXakX/BXya',
    },
    rating: 4.8,
    reviewsCount: 230,
  },
  {
    id: '2',
    title: 'Macchiato',
    extras: 'Oat Milk',
    description:
      'An espresso-based drink ingredients a small amount of frothed milk, offering a bolder coffee flavor ingredients a touch of creaminess.',
    price: 5.29,
    image: {
      source:
        'https://images.unsplash.com/photo-1473923377535-0002805f57e8?q=80&w=3108&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: 'kBgOHIIIl3h5iHdziogId4KBCA',
    },
    rating: 4.9,
    reviewsCount: 230,
  },
  {
    id: '3',
    title: 'Latte',
    extras: 'Chocolate',
    description:
      'A creamy espresso-based drink ingredients steamed milk, known for its smooth texture.',
    price: 5.39,
    image: {
      source:
        'https://images.unsplash.com/photo-1598971948291-38c380f80f85?q=80&w=2970&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: '5HoKFYJ/uHZ7iHdmeoV4ecVqL3v2',
    },
    rating: 4.8,
    reviewsCount: 230,
  },
  {
    id: '4',
    title: 'Americano',
    extras: 'hot water',
    description:
      'An Americano is made by diluting espresso ingredients hot water, resulting in a similar strength to regular coffee but ingredients a distinct espresso flavor profile.',
    price: 4.99,
    image: {
      source:
        'https://images.unsplash.com/photo-1582252852999-5ca546037481?q=80&w=2970&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: '1zgKDYJ2d5aviHdmerhXakX/BXya',
    },
    rating: 4.9,
    reviewsCount: 230,
  },
  {
    id: '5',
    title: 'Cappuccino',
    extras: 'foamed milk',
    description:
      'An espresso-based drink ingredients equal parts of espresso, steamed milk, and foamed milk, offering a balance of flavors and textures.',
    price: 4.53,
    image: {
      source:
        'https://images.unsplash.com/photo-1472973681244-f5bcc808ad47?q=80&w=2974&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: '1zgKDYJ2d5aviHdmerhXakX/BXya',
    },
    rating: 4.8,
    reviewsCount: 230,
  },
  {
    id: '6',
    title: 'Machiato',
    extras: 'Oat Milk',
    description:
      'An espresso-based drink ingredients a small amount of frothed milk, offering a bolder coffee flavor ingredients a touch of creaminess.',
    price: 4.53,
    image: {
      source:
        'https://images.unsplash.com/photo-1473923377535-0002805f57e8?q=80&w=3108&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
      thumbhash: 'kBgOHIIIl3h5iHdziogId4KBCA',
    },
    rating: 4.9,
    reviewsCount: 230,
  },
];

const cart = [
  {
    id: 1,
    items: [
      {
        item_id: 101,
        name: 'Espresso',
        quantity: 2,
        size: 'Regular',
      },
      {
        item_id: 102,
        name: 'Cappuccino',
        quantity: 1,
        size: 'Large',
      },
    ],
    total_price: 9.5,
  },
  {
    id: 2,
    items: [
      {
        item_id: 103,
        name: 'Latte',
        quantity: 1,
        size: 'Regular',
      },
    ],
    total_price: 3.75,
  },
  {
    id: 3,
    customer_name: 'Alice Johnson',
    items: [
      {
        item_id: 104,
        name: 'Mocha',
        quantity: 1,
        size: 'Large',
      },
      {
        item_id: 105,
        name: 'Croissant',
        quantity: 2,
        size: 'Small',
      },
    ],
    total_price: 14.8,
  },
];

const notifications = [
  {
    content: '3x Espresso',
    id: '3',
    read: true,
    timestamp: '2023-12-04T15:30:00Z',
    title: 'Order #1236',
  },
  {
    content: '2x Latte, 1x Croissant',
    id: '1',
    read: false,
    timestamp: '2023-12-05T08:00:00Z',
    title: 'Order #1234',
  },
  {
    content: '1x Cappuccino, 1x Blueberry Muffin',
    id: '2',
    read: true,
    timestamp: '2023-12-06T10:00:00Z',
    title: 'Order #1235',
  },
  {
    content: '2x Americano, 2x Chocolate Croissant',
    id: '4',
    read: false,
    timestamp: '2023-12-07T09:45:00Z',
    title: 'Order #1237',
  },
  {
    content: '1x Macchiato, 1x Scone',
    id: '5',
    read: false,
    timestamp: '2023-12-08T11:20:00Z',
    title: 'Order #1238',
  },
  {
    content: '4x Flat White',
    id: '6',
    read: false,
    timestamp: '2023-12-09T07:55:00Z',
    title: 'Order #1239',
  },
  {
    content: '2x Mocha',
    id: '7',
    read: true,
    timestamp: '2023-12-10T14:00:00Z',
    title: 'Order #1240',
  },
  {
    content: '1x Espresso, 1x Danish Pastry',
    id: '8',
    read: false,
    timestamp: '2023-12-11T10:30:00Z',
    title: 'Order #1241',
  },
  {
    content: '3x Iced Coffee',
    id: '9',
    read: false,
    timestamp: '2023-12-12T08:45:00Z',
    title: 'Order #1242',
  },
  {
    content: '1x Pumpkin Spice Latte',
    id: '10',
    read: true,
    timestamp: '2023-12-13T16:20:00Z',
    title: 'Order #1243',
  },
];

export { cart, categories, coffees, notifications };
