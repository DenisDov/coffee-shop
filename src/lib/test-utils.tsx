import {
  DarkTheme,
  ThemeProvider as NavigationThemeProvider,
} from '@react-navigation/native';
import { ThemeProvider } from '@shopify/restyle';
import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { GestureHandlerRootView } from 'react-native-gesture-handler';

import { theme } from '@/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <ThemeProvider theme={theme}>
        <NavigationThemeProvider value={DarkTheme}>
          {children}
        </NavigationThemeProvider>
      </ThemeProvider>
    </GestureHandlerRootView>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
