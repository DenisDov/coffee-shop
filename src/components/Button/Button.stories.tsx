import { Meta, StoryObj } from '@storybook/react-native';
import React from 'react';
import { View } from 'react-native';

import { Button, ButtonProps } from './Button';

const meta: Meta<ButtonProps> = {
  title: 'components/Button',
  component: Button,
  argTypes: {
    onPress: {
      action: 'onPress event',
    },
  },
  decorators: [
    Story => (
      <View style={{ padding: 16 }}>
        <Story />
      </View>
    ),
  ],
};

export default meta;

type Story = StoryObj<ButtonProps>;

export const Basic: Story = {
  args: {
    // disabled: false,
    text: 'Tap me',
  },
};

export const Disabled: Story = {
  args: {
    // disabled: true,
    text: 'Disabled',
  },
};
