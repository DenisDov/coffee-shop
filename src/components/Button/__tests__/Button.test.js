import React from 'react';

import { fireEvent, render } from '@/lib/test-utils';

import { Button } from '../Button';

describe('Button Component', () => {
  test('renders button with correct text', () => {
    const onPressMock = jest.fn();
    const buttonText = 'Click me';

    const { getByText } = render(
      <Button onPress={onPressMock} text={buttonText} />,
    );

    const button = getByText(buttonText);
    expect(button).toBeTruthy();
  });

  test('calls onPress when button is pressed', () => {
    const onPressMock = jest.fn();
    const buttonText = 'Click me';

    const { getByText } = render(
      <Button onPress={onPressMock} text={buttonText} />,
    );

    const button = getByText(buttonText);
    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
