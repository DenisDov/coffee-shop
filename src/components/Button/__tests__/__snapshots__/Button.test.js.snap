// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with correct text 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  delayLongPress={600}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  hitSlop={16}
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[MockFunction]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#C67C4E",
      "borderRadius": 16,
      "height": 64,
      "justifyContent": "center",
      "paddingHorizontal": 8,
    }
  }
  underlayColor="black"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "black",
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderRadius": 16,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    accessibilityRole="button"
    accessible={true}
    style={
      [
        {},
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "Sora_600SemiBold",
            "fontSize": 16,
            "textTransform": "capitalize",
          },
        ]
      }
    >
      Click me
    </Text>
  </View>
</RNGestureHandlerButton>
`;
